### DISCUSSION TITLE
Handling multiple sys._meipass directories for a one-file bundle

### DISCUSSION THREAD

**Discussion by eliranma** [pyinstaller/pyinstaller#7952] [Help]
Hi everyone.
I'm using the library for many of my apps, and i saw this section about the one-file bundle.
if a user shutdown his machine, or have to force-kill, any way to find the last used directory and use it, instead of creating new one?
or maybe to remove all the old directories.

```
After creating the temporary folder, the bootloader proceeds exactly as for the one-folder bundle, in the context of the temporary folder. When the bundled code terminates, the bootloader deletes the temporary folder.

(In GNU/Linux and related systems, it is possible to mount the /tmp folder with a “no-execution” option. That option is not compatible with a PyInstaller one-file bundle. It needs to execute code out of /tmp. If you know the target environment, [--runtime-tmpdir](https://pyinstaller.org/en/stable/usage.html#cmdoption-runtime-tmpdir) might be a workaround.)

Because the program makes a temporary folder with a unique name, you can run multiple copies of the app; they won’t interfere with each other. However, running multiple copies is expensive in disk space because nothing is shared.

The _MEIxxxxxx folder is not removed if the program crashes or is killed (kill -9 on Unix, killed by the Task Manager on Windows, “Force Quit” on macOS). Thus if your app crashes frequently, your users will lose disk space to multiple _MEIxxxxxx temporary folders.

It is possible to control the location of the _MEIxxxxxx folder by using the [--runtime-tmpdir](https://pyinstaller.org/en/stable/usage.html#cmdoption-runtime-tmpdir) command line option. The specified path is stored in the executable, and the bootloader will create the _MEIxxxxxx folder inside of the specified folder. Please see [Defining the Extraction Location](https://pyinstaller.org/en/stable/usage.html#defining-the-extraction-location) for deta
```

**Comment by bwoodsend** [comment]
I think this goes under the same *it's more trouble than it's worth* umbrella as https://github.com/pyinstaller/pyinstaller/issues/7907.

To be clear, this only applies to the very aggressive -9 signal (and whatever the Windows equivalent is). It's not something that will happen frequently. And I'm fairly certain that the bit about shutdown is inaccurate – all OSs I know of will try to politely terminate all running applications before shutting down.

**Reply by eliranma** [reply]
If the user close the app using ❌ button that will call `sys.exit(0)` the next run should go to the same directory?

**Reply by bwoodsend** [reply]
No, the directory will be deleted as the process exits because `sys.exit()` is not one of those few abnormal error types (in fact it's not even an error) that lead to process's cleanup code being bypassed. 

The directory will never be reused no matter how it exits.
